core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -84,6 +84,7 @@ public static Codec fromFileName(String fileName) {
   private TableMetadataParser() {}
 
   // visible for testing
+  static final String METADATA_FILE_LOCATION = "metadata-file-location";
```

Comment:
this isn't part of the [Spec](https://iceberg.apache.org/spec/#table-metadata), so I don't think we can just add it here